rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -1459,7 +1459,7 @@ data class ForStmt(
 @Serializable
 data class SortAStmt(val target: Expression, override val position: Position? = null) : Statement(position) {
     override fun execute(interpreter: InterpreterCore) {
-        sortA(interpreter.eval(target), interpreter.getLocalizationContext().charset)
+        sortA(interpreter.eval(target), target.type(), interpreter.getLocalizationContext().charset)
```

Comment:
In kotlin fortunatly you can pass the parameter by using also their name, force himself to use this feature because is very helpful when, as you made in the sortA implementation, you add furthers parameters to the function signature